### ISSUE TITLE
update on project priority and plans for next steps

### ISSUE THREAD

**Issue by ild0tt0re** [Shopify/slate#1063]
Dear Shopify Team, from your last announcement

> This low maintenance period is time-boxed to 6 months from March 2019. We will re-visit and re-evaluate the project priority and plans for next steps.

More than six months have passed and we would like to hear about this project as we are evaluating whether to use Shopify platform for our customers.

We consider this project of primary importance.
Thanks.

**Comment by javy-** [comment]
@t-kelly is the version bump indicative of development resuming?

https://github.com/Shopify/slate/commit/d3b4cd867f91d0c8b56c61f438c096090dee1c59

**Comment by t-kelly** [comment]
No -- that was only me jumping in for some quick wins. Still waiting on the official word of how things will proceed.

**Comment by lmartins** [comment]
I think it needs to be said, that the way Shopify is treating the developers who invest and work with this platform is wildly disrespectful here. I'm sure there are bigger reasons here, and perhaps something good is coming down the road, but no way this is a proper response to people who help build the platform.

**Comment by joeldmyers** [comment]
I agree, and I think Shopify does need to consider whether it's worth investing in its development community.  Developers are often the ones deciding which platform and technologies their clients should choose, and to be totally honest Shopify doesn't have a great reputation among developers right now.  Part of that is ignorance as to how robust the platform can be.  But part of that is the direct result of this kind of neglect, and not keeping apace with state of the art development.  It definitely does not reflect well on what's an approx. $34bn company, that their primary way of making custom themes seems to have been abandoned, and Shopify will definitely lose a lot of money in the long run from developers who decide to choose other platforms for their clients, if there isn't adequate support and tooling available.

**Comment by cgilbe27** [comment]
You don't necessarily have to use slate; to use npm packages, CI/CD, or linting. We have accomplished all of it. Slate mainly provides direct tooling with Shopify along with some built in linting support and deploying, nothing unique by no means.
In its current state, Slate is not usable for full development, you can use it, but don't rely on it. Also, the secure localhost issue was a huge problem we ran into, which was the main reason we went with a piecemeal solution for a better comprehensive solution, and dropped Slate completely.

**Comment by joeldmyers** [comment]
Ah yeah makes sense; I've put in a lot of additional things (linting, E2E / unit tests etc.) but kept Slate for deployment... I might just need to do the same and build something out for deployment on my own.  Thanks!

**Comment by cgilbe27** [comment]
I strongly recommend deployhq, awesome deploy service and great technical support

**Comment by lmartins** [comment]
@cgilbe27 while I'm curious as to how you use deployhq to achieve an equivalent workflow allowed by slate (livereload, deployments, etc.) lets perhaps keep this issue on topic and that is, the lack of information coming from Shopify on the future of this project.

**Comment by curiouscrusher** [comment]
While it's clear we all want to have some insight on what the future of Shopify's developer offerings will be (Both with Slate and other toolsets), it's important to keep a few things in mind. 

First, Shopify's engineering staff more than likely does not get to set the priorities for company wide initiatives. So while venting into a GitHub issue on an open source project might make you feel better, it's highly unlikely to truly accomplish anything useful at the company level. 
Secondly, this is an open source project, it's provided as-is with the intention of helping developers who may not have time/resources to hand-roll their own solution. It's also a great learning tool that can give insight into how the Shopify team builds and architects tools/features. 

If you're unhappy with Slate, there's a very large list of alternatives out there. And if none of those meet your needs then I suggest you build your own development workflow and tooling, that's what plenty of others have done.

**Comment by cgilbe27** [comment]
@lmartins Basically my point was that while Slate has features, they are definitely not hashed out completely, (I mean just look at the amount of issue requests), so from an objective point of view, if you want features outside of Slate or something in Slate that isnt broken, you'll need to tie those in yourself, but that its very simple to do so. OP stated they were weighing whether or not to go with Shopify, but I dont think Slate should be a reason to not go with Shopify for your site.
Also, themekit and prepros does great for JS and SCSS minification + it includes browser reload and is suggested by Shopify.
An update would be nice however, because we would likely want to extend Slate in the future once its at a non-beta stage.

**Comment by scottsb** [comment]
@curiouscrusher Venting here may not accomplish much, but this issue has visibility up to the CEO:

June: https://twitter.com/tobi/status/1138581134156161025
November: https://twitter.com/tobi/status/1191715755127136256

**Comment by drinkmorewaters** [comment]
@t-kelly Just to say, that the only reason i can take any enjoyment developing for Shopify is Slate. Even its not perfect, having a local development is just so much better than using the Shopify editor. We use Vue a lot so its working ok. Any modern development pipeline is appreciated and the work put in doesn't go without appreciation.

So, thank you!

**Comment by amosmos** [comment]
Hey all and especially @t-kelly,

Just wanted to say how great Slate is already now, and how great it was to transfer the SodaStream set of Shopify stores to Slate.

We launched it today and it's already showing amazing performance results!

Thanks!!!

**Comment by chanmathew** [comment]
What's everyone using now that Slate is in low maintenance? I've only found Slater so far....are there other alternatives I'm not aware of?

**Comment by curiouscrusher** [comment]
@chanmathew I’ve explored slater as well as some other options like Motifmate and even some of the Plus Partners house rolled solutions. I always end up back at Slate because it’s just so well rounded and fits all the needs of flexible Shopify theme development.
Even on low maintenance it’s very much a complete solution, and since it’s open source you can customize the code to suit your needs. Most of the other solutions are too targeted or opinionated for my needs and I’ll end up spending more time customizing them than I would have spent just starting with Slate.

**Comment by aeu** [comment]
It is now January 2020, nine months since the 6 month time-boxed period. Is there any update available regarding the status of this project?

**Comment by adrianocr** [comment]
I'm a contractor who does work for 3 prominent marketing companies whose clients are basically 99% shopify stores and people looking to build ecommerce stores (which we end up building with shopify). MANY of which are Shopify Plus stores. 

I am now in the process of considering alternatives and suggesting other platforms if this is how we're going to be treated as developers. This _was_ a nice symbiotic relationship between Shopify and developers but it seems that one half of that partnership has decided it is no longer interested...

**Comment by javy-** [comment]
[Slate - End of Support (January 2020)](https://github.com/Shopify/slate/commit/1717130d45bc0166161c828708270dc7269f3ebf)

For anyone following this thread who didn't see the update yesterday...

**Comment by drinkmorewaters** [comment]
Thank goodness Shopify finally had the (Apple courage) to come out and put Slate down.

Google EOL meet Shopify EOL. Two pees in a pod.

**Comment by lmartins** [comment]
Agreed. Shame how long it took for them to come clear on this,  but glad they finally did it. I don't think at this stage these news have caught anyone by surprise.

**Comment by adrianocr** [comment]
Ok so you guys say that Slate doesn't solve the problem of local development and theme versioning. Cool, I agree. But is that that? Or do you guys have any plans to do something about it? Because your statement basically comes off like "Yeah it doesn't work how we'd like so we're ending support. Good luck." and that's it.

Are you guys happy enough with the offerings from Maestroo and Pixel Union that you think they fit every customer's needs? I think it's ill-advised to alienate your developers like this. There's what, a few hundred of us? Few thousand? Are we not enough of a financial incentive at Shopify that you guys feel you can ignore us?

I've personally built 52 Shopify stores to date and about ~48 of those were custom themes. 12 of them Shopify Plus stores. Many times we end up pushing the customer towards Shopify because we like the platform. But if Slate is no longer supported then building themes on Shopify becomes archaic again and we have little to no incentive to not go with WooCommerce where yeah, WordPress has its pitfalls but at least building themes for it is a modern and easy experience. Bigcommerce has its issues but at least building themes for it is a modern experience.

Yes I'm being hyperbolic here but what I'm trying to get at is: how about some better communication? How about a ballpark road map? Hell, I'd even take a ballpark for an announcement date for a road map at this point.

**Comment by lmartins** [comment]
@adrianocr I've been where you at. Super frustrated with how they have treated developers here. I'm done with that.

They are saying that you can use themekit and build on top of that. Personally I'm using theme kit and my own Webpack config to achieve something that fits my needs. Maybe this whole situation with spark the appearance of new tooling, and maybe Shopify will have something new in the future. Either way, the way they have conducted this was very poorly managed.

**Comment by curiouscrusher** [comment]
Well that sucks, far more drawn out than it should have been. And I can’t say either of the stated reasons for sundown even seem related to Slate all that much. 

- Local Theme Dev: that’d be cool, but honestly I’ve never had an issue with proxying the live store. Is it subpar? Yeah sure, but it *worked*. This has very little if anything to do with Slate, and everything to do with the Shopify platform. Your build process shouldn’t necessarily be responsible for local emulation. Truly local theme development would need to account for unique store data, customers, products, Metafields, Apps, Checkout, basically an entire store. That seems way out of scope for a theme building tool. 
- Theme Versioning: Again, this is a platform problem. And definitely a real issue, but still seems like a stretch to consider it part of theme development. It’s certainly more related than local development, but it’s really just a platform limitation that there’s no proper theme versioning system outside of duplicating a theme. 

Seems like the vision is far overreaching the actual need and problems that ahould be solved. A simple, reliable, and maintained theme development tool would likely go much further with the community than some grand plan to unify local development, new sections, and theme versioning. As much as I’d love to see proper version control for Shopify Themes, we really just need a steady tool to work with for longer than 6-12 months. Especially if Shopify expects the development community to grow, or even stay steady.

**Comment by drinkmorewaters** [comment]
> @adrianocr I've been where you at. Super frustrated with how they have treated developers here. I'm done with that.
> 
> They are saying that you can use themekit and build on top of that. Personally I'm using theme kit and my own Webpack config to achieve something that fits my needs. Maybe this whole situation with spark the appearance of new tooling, and maybe Shopify will have something new in the future. Either way, the way they have conducted this was very poorly managed.

It would be ill-advised to build your own tool set for stores that have limited management, considering theme-kit, could be EOL at any stage, and the new theme architecture could render whatever tooling you build useless at any point.

**Comment by lmartins** [comment]
> It would be ill-advised to build your own tool set for stores that have limited management, considering theme-kit, could be EOL at any stage, and the new theme architecture could render whatever tooling you build useless at any point.

Yes, Shopify could pull the same move on Themekit and that would suck. But to say that it would be ill-advised to know your tools and explore Webpack to serve your purpose feels a fit of a stretch. You can set it to have hot reloading for JS and SASS/CSS, and you probably can setup Browsersync to do the same for liquid files. 

Am I happy with how things unfolded, absolutely not. I've expressed many times on this forum how disrespectful this whole ordeal was towards developers, and aside from @t-kelly who gracefully tried to give us some context, I haven't seen any good reasoning behind this. Much to the contrary, as @curiouscrusher mentions, this decision only contradicts their arguments to get slate on a low maintenance in the first place.

With all that said, its the web, you have options, both on tooling and on e-commerce platforms too.

**Comment by drinkmorewaters** [comment]
> > It would be ill-advised to build your own tool set for stores that have limited management, considering theme-kit, could be EOL at any stage, and the new theme architecture could render whatever tooling you build useless at any point.
> 
> Yes, Shopify could pull the same move on Themekit and that would suck. But to say that it would be ill-advised to know your tools and explore Webpack to serve your purpose feels a fit of a stretch. You can set it to have hot reloading for JS and SASS/CSS, and you probably can setup Browsersync to do the same for liquid files.
> 
> Am I happy with how things unfolded? Absolutely not!
> I've expressed many times on this forum how disrespectful this whole ordeal was towards developers, and aside from @t-kelly who gracefully tried to give us some context, I haven't seen any good reasoning behind this. Much to the contrary, as @curiouscrusher mentions, this decision only contradicts their arguments to get slate on a low maintenance in the first place.
> 
> With all that said, it's the web, you have options, both on tooling and on e-commerce platforms too.

What i meant, and it still stands, is that it is ill advised to build tools for clients of which you won't maintain. So building anything for a client where in 6 months, Shopify could kill it, is and always will be ill-advised. 

From this day forward, if you use Slate for a client, that's ill advised, if you build something custom for a client that you won't be maintaining that tooling for, that's ill advised.

And if you trust Shopify to build standard, consistent, reliable tooling, with great documentation, at the level of something such as VueJS or React, that's ill advised.

**Comment by lmartins** [comment]
@drinkmorewaters and you're still not getting what I meant but I couldn't care less. Drink more water brother.

**Comment by drinkmorewaters** [comment]
You have no point, this is about Slate, go learn webpack somewhere else, for commercial use, don't setup clients for projects that get EOL or could break at any time of which, they can't afford to maintain. If building a toolset for Shopify is easy for you, you take over Slate.

> But to say that it would be ill-advised to know your tools and explore Webpack to serve your purpose feels a fit of a stretch.

**Comment by lmartins** [comment]
@drinkmorewaters all I said is that if you do this for a living, you have alternatives and you can kinda make it work until something better comes up. I'm not here to learn Webpack, I know where to look for help on that and I'm as entitled as you to come here for help on Slate.

What I did was offering a different perspective, something that you seem to have a problem with. That tells more about you than it does about.

Whatever tool you use to upload, the end product is the same, a damn theme you can take from there any way you like. If that doesn't work for you, then yeah, go cry in front of Shopify headquarters or something, because that's what you have now.

**Comment by adrianocr** [comment]
Maybe you two should go argue in private.

The point stands that at this point Shopify still refuses to communicate with the developers who pour hours upon hours upon hours developing themes and stores for their platform, refuses to give us even a hint at future plans, a road map, and is all around jsut treating us like an unimportant cog in the wheel. Do they think they're too big to fail?

And yeah themekit works but for how long? Next thing we know that starts throwing out errors, issues pile up in that repo, and we get another EOL announcement.
